### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Oleh-Khashchevskyi**
[DEMO LINK](https://Oleh-Khashchevskyi.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

File: src/utilities/Validation.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  value: true,
+  message: 'This field is required',
+};
+
+export const isEmailValid = {
+  value: /\S+@\S+\.\S+/,
+  message: 'E-mail is not valid',
+};
+
+export const isMinValue = (value:string | undefined, min:number) => {
```
Comment (TarasHoliuk) [inline src/utilities/Validation.ts:11]:
```suggestion
export const isMinValue = (value?: string, min:number) => {
```

File: src/utilities/HttpClient.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import axios from 'axios';
+
+const instance = axios.create({
+  baseURL: 'https://mate.academy/students-api',
+});
+
+export const HttpClient = {
```
Comment (TarasHoliuk) [inline src/utilities/HttpClient.ts:7]:
```suggestion
export const httpClient = {
```
use camelCase if it's not React component/type/class etc.

File: src/services/StorageService.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import User from 'models/User';
+
+const StorageService = {
```
Comment (TarasHoliuk) [inline src/services/StorageService.ts:3]: camelCase

File: src/store/users/usersSlice.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  name: 'users',
+  initialState,
+  reducers: {
+    setCurrentUser: (state:State, action:PayloadAction<User | null>) => {
+      state.currentUser = action.payload;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      // Fetch Users
```
Comment (TarasHoliuk) [inline src/store/users/usersSlice.ts:24]: I guess this comment is redundant - it is clear from the code

File: src/components/UserMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          >
+            {currentUser.name}
+            <ArrowDropDownIcon />
+          </Box>
+        )}
+        onClick={openMenu}
+      />
+      <Menu>
+        <MenuItem onClick={handleClick}>
+          Sign Out
```
Comment (TarasHoliuk) [inline src/components/UserMenu.tsx:42]: :+1:

**Comment by TarasHoliuk** [comment]
probably, you have to install smth like eslint-plugin-import to fix it (or just use relative paths):
image

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
Nicely done :+1: 
You've done all 3 parts in the first task with additional futures :fire: 

I've left a few small comments

File: src/utilities/Validation.ts (hunk 1 above)
Comment (Oleh-Khashchevskyi) [reply src/utilities/Validation.ts:11]: cannot be set this way

unless so (value: string = '', min:number)

File: src/utilities/Validation.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  value: true,
+  message: 'This field is required',
+};
+
+export const isEmailValid = {
+  value: /\S+@\S+\.\S+/,
+  message: 'E-mail is not valid',
+};
+
+export const isMinValue = (value:string, min:number) => {
```
Comment (tetiana-ivanchenko) [inline src/utilities/Validation.ts:11]:
```suggestion
export const isMinValue = (value: string, min: number) => {
```

**Review by tetiana-ivanchenko — APPROVED** [review]

**Pull request closed without merging**